src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -832,5 +836,376 @@ private static JsonSerializerOptions CreateStringEnumOptionsForType(Type enumTyp
+
+                [EnumFlagsWithMemberAttributes.Value1 | EnumFlagsWithMemberAttributes.Value4] = EnumFlagsWithMemberAttributes.Value3,
+                [EnumFlagsWithMemberAttributes.Value4] = EnumFlagsWithMemberAttributes.Value2,
+            };
+
+            string json = JsonSerializer.Serialize(value, options);
+            JsonTestHelper.AssertJsonEqual("""
+                {
+                    "A": "A, B, C, value4",
+                    "A, VALUE4": "C",
+                    "VALUE4": "B"
+                }
+                """, json);
+        }
+
+        [Theory]
+        [InlineData("\"a\"")]
+        [InlineData("\"b\"")]
+        [InlineData("\"A, b\"")]
+        [InlineData("\"A, b, C, Value4\"")]
+        [InlineData("\"A, B, c, Value4\"")]
+        [InlineData("\"a, b, c, Value4\"")]
+        [InlineData("\"c, B, A, Value4\"")]
+        public static void EnumFlagsWithMemberAttributes_CustomizedValuesAreCaseSensitive(string json)
+        {
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<EnumFlagsWithMemberAttributes>(json));
+        }
+
+        [Theory]
+        [InlineData("\"value4\"", EnumFlagsWithMemberAttributes.Value4)]
+        [InlineData("\"value4, VALUE4\"", EnumFlagsWithMemberAttributes.Value4)]
+        [InlineData("\"A, value4, VALUE4, A,B,A,A\"", EnumFlagsWithMemberAttributes.Value1 | EnumFlagsWithMemberAttributes.Value2 | EnumFlagsWithMemberAttributes.Value4)]
+        [InlineData("\"VALUE4, VAlUE5\"", EnumFlagsWithMemberAttributes.Value4 | EnumFlagsWithMemberAttributes.Value5)]
+        public static void EnumFlagsWithMemberAttributes_DefaultValuesAreCaseInsensitive(string json, EnumFlagsWithMemberAttributes expectedValue)
+        {
+            EnumFlagsWithMemberAttributes value = JsonSerializer.Deserialize<EnumFlagsWithMemberAttributes>(json);
+            Assert.Equal(expectedValue, value);
+        }
+
+        [Flags, JsonConverter(typeof(JsonStringEnumConverter<EnumFlagsWithMemberAttributes>))]
+        public enum EnumFlagsWithMemberAttributes
+        {
+            [JsonStringEnumMemberName("A")]
+            Value1 = 1,
+            [JsonStringEnumMemberName("B")]
+            Value2 = 2,
+            [JsonStringEnumMemberName("C")]
+            Value3 = 4,
+            Value4 = 8,
+            Value5 = 16,
+        }
+
+        [Theory]
+        [InlineData(EnumWithConflictingMemberAttributes.Value1)]
+        [InlineData(EnumWithConflictingMemberAttributes.Value2)]
+        [InlineData(EnumWithConflictingMemberAttributes.Value3)]
+        public static void EnumWithConflictingMemberAttributes_IsTolerated(EnumWithConflictingMemberAttributes value)
+        {
+            string json = JsonSerializer.Serialize(value);
+            Assert.Equal("\"Value3\"", json);
+            Assert.Equal(EnumWithConflictingMemberAttributes.Value1, JsonSerializer.Deserialize<EnumWithConflictingMemberAttributes>(json));
+        }
```

Comment:
Should we throw for this case instead?

Reply:
I considered it, but this isn't the case today when you have conflicts via a naming policy so I figured we should prefer consistency.

Reply:
It is the case for conflicting [JsonPropertyName]s, though.

Reply:
True, properties contain payloads however so duplication usually results in serialization errors downstream. Enums OTOH are a bit more flexible in that regard, the language even lets you reuse the same underlying value for multiple fields:
```C#
enum Foo { A = 1, B = 1 }
```